### PR TITLE
Superpixel Segmentation CLI

### PR DESCRIPTION
**Pull request by slee172**
This PR makes a superpixel segmentation CLI that creates two files (features and boundaries) used as a dataset for HistomicsML.

For boundaries, compute_boundary_data function works on in parallel using Dask and extracts superpixel boundaries from WSIs.

For features, compute_superpixel_data function works on CPU/GPU and extracts features for each superpixel patch.

CLI input is a directory of WSIs and a fitted PCA model and CLI output is two files(features and boundaries) which are used as a dataset for HistmocsML.

### CODE REVIEW

File: histomicstk/cli/SuperpixelSegmentation/CreateDataset.py [hunk 1]
```diff
@@ -0,0 +1,615 @@
+import os
```
Comment (cdeepakroy) [inline histomicstk/cli/SuperpixelSegmentation/CreateDataset.py]: @slee172 

The name of the CLI's python/xml file and its parent directory should match.

File: histomicstk/cli/SuperpixelSegmentation/CreateDataset.xml [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+      <description>Standard deviation of reference image in LAB color space for Reinhard color normalization</description>
+      <longflag>reference_std_lab</longflag>
+      <default>0.57506023, 0.10403329, 0.01364062</default>
+    </double-vector>
+    <double-vector>
+      <name>source_mu_lab</name>
+      <label>Source Mean LAB</label>
+      <description>Mean of source image in LAB color space for Reinhard color normalization</description>
+      <longflag>source_mu_lab</longflag>
+      <default>8.96447078, -0.0879669, 0.04412108</default>
```
Comment (cdeepakroy) [inline histomicstk/cli/SuperpixelSegmentation/CreateDataset.xml]: Source mu and std in Lab is supposed to computed from the slide being processed.

You can provide these as parameters but then the default values should be some None value like -1, -1, -1. If these parameters are not supplied then it should compute them on the fly.

File: histomicstk/cli/SuperpixelSegmentation/CreateDataset.xml [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+      <default>5</default>
+    </double>
+    <double>
+      <name>superpixel_tile_size</name>
+      <label>Superpixel Analysis Tile Size</label>
+      <description>Tile size for blockwise analysis</description>
+      <longflag>superpixel_tile_size</longflag>
+      <default>4096</default>
+    </double>
+    <double>
```
Comment (cdeepakroy) [inline histomicstk/cli/SuperpixelSegmentation/CreateDataset.xml]: Add super pixel parameters into a separate parameter group.

File: histomicstk/cli/SuperpixelSegmentation/CreateDataset.xml [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+      <name>analysis_mag</name>
+      <label>Analysis Magnification</label>
+      <description>The magnification at which the analysis should be performed.</description>
+      <longflag>analysis_mag</longflag>
+      <default>5</default>
+    </double>
+    <double>
+      <name>superpixel_tile_size</name>
```
Comment (cdeepakroy) [inline histomicstk/cli/SuperpixelSegmentation/CreateDataset.xml]: why rename this with "superpixel_" prefix?

**Comment by slee172** [comment]
Generating a superpixel dataset is already implemented by using a docker image, so I closed this PR.

**Pull request closed without merging**